files.go

```diff
@@ -0,0 +1,89 @@
+package iface
+
+import (
+	"context"
+	"io"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-cidutil/cidenc"
+	"github.com/ipfs/go-mfs"
+)
+
+type FilesCopyOptions struct {
+	Flush bool
+}
+
+type FilesMoveOptions struct {
+	Flush bool
+}
+
+type FilesListOptions struct {
+	CidEncoder cidenc.Encoder
+	Long       bool
+}
+
+type FilesRemoveOptions struct {
+	Force     bool
+	Recursive bool
+}
+
+type FilesStatOptions struct {
+	CidEncoder   cidenc.Encoder
+	WithLocality bool
+}
+
+type FilesWriteOptions struct {
+	Create            bool
+	MakeParents       bool
+	Truncate          bool
+	Flush             bool
+	RawLeaves         bool
+	RawLeavesOverride bool
+	Offset            int64
+	// Count is the number of bytes to write. 0 (default) writes everything.
+	Count      int64
+	CidBuilder cid.Builder
+}
+
+type FilesReadOptions struct {
+	Offset int64
+	// Count is the number of bytes to read. 0 (default) reads everything.
+	Count int64
+}
+
+type FilesMkdirOptions struct {
+	MakeParents bool
+	Flush       bool
+	CidBuilder  cid.Builder
+}
+
+type FilesChangeCidOptions struct {
+	Flush      bool
+	CidBuilder cid.Builder
+}
+
+type FileInfo struct {
+	Hash           string
+	Size           uint64
+	CumulativeSize uint64
+	Blocks         int
```

Comment:
I have no idea why we called this "blocks". Let's call it what it is, Children (NumChildren?).